components/library/DateRangePicker/Preview/DateRangePicker.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import {
+  Popover,
+  Button,
+  TextField,
+  Box,
+  DatePicker,
+  Icon,
+  OptionList,
+  Scrollable,
+  InlineGrid,
+  BlockStack,
+  InlineStack
+} from '@shopify/polaris';
+import { ArrowRightIcon, CalendarIcon } from '@shopify/polaris-icons';
+
+export const DateRangePicker = ({ onDateRangeSelect }) => {
+  const today = new Date(new Date().setHours(0, 0, 0, 0));
+  const yesterday = new Date(
+    new Date(new Date().setDate(today.getDate() - 1)).setHours(0, 0, 0, 0)
+  );
+
+  const ranges = [
+    { title: 'Today', period: { since: today, until: today } },
+    { title: 'Yesterday', period: { since: yesterday, until: yesterday } },
+    {
+      title: 'Last 7 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 7)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 30 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 30)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 90 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 90)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 365 Days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 365)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Custom',
+      period: { since: yesterday, until: yesterday }
+    }
+  ];
+
+  const [popoverActive, setPopoverActive] = useState(false);
+  const [activeDateRange, setActiveDateRange] = useState(ranges[0]);
+  const [dateState, setDateState] = useState({
+    month: activeDateRange.period.since.getMonth(),
+    year: activeDateRange.period.since.getFullYear()
+  });
+
+  useEffect(() => {
+    setDateState({
+      month: activeDateRange.period.since.getMonth(),
+      year: activeDateRange.period.since.getFullYear()
+    });
+  }, [activeDateRange]);
+
+  const handleInputValueChange = useCallback((field, value) => {
+    if (isValidDate(value)) {
+      const newDate = new Date(value);
+      setActiveDateRange((prevState) => ({
+        ...prevState,
+        period: {
+          ...prevState.period,
+          [field]: newDate
+        }
+      }));
+    }
+  }, []);
+
+  function formatDate(date) {
```

Comment:
If you could update these to be arrow functions that'd be great, just for consistency throughout the other components